src/components/Request.js

```diff
@@ -97,7 +97,14 @@ class Request extends React.Component {
 
   render() {
     if (typeof this.props.children === 'function') {
-      return this.props.children(this.state.error, this.state.response, this.state.isLoading, (props) => this.onReload(props))
+      const _axios = this.props.instance || this.context.axios || axios
+      return this.props.children(
+        this.state.error,
+        this.state.response,
+        this.state.isLoading,
+        (props) => this.onReload(props),
+        _axios
```

Comment:
@sheaivey ok, if I pass `this.makeRequest` as well? It might be usefull to use `makeRequest` in wrapped component, as wrapped might need to make request at specific point (i.e. when user submit the form). So, without passed `makeRequest`, user can't build **stateless** component.

Reply:
I think `onReload()` is essentially `makeRequest()` but with a debounce.

Also, any time options.url` changes while `options.isReady` = `true` a makeRequest() will also get triggered.

Reply:
Maybe the naming of the prop could be better though?

Reply:
@sheaivey  you seems to already covered all the cases, great job! I hardly can see what I can add :)